x-pack/plugins/apm/public/components/shared/span_icon/get_span_icon.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import awsIcon from './icons/aws.svg';
+import cassandraIcon from './icons/cassandra.svg';
+import databaseIcon from './icons/database.svg';
+import defaultIconImport from './icons/default.svg';
+import documentsIcon from './icons/documents.svg';
+import elasticsearchIcon from './icons/elasticsearch.svg';
+import globeIcon from './icons/globe.svg';
+import graphqlIcon from './icons/graphql.svg';
+import grpcIcon from './icons/grpc.svg';
+import handlebarsIcon from './icons/handlebars.svg';
+import kafkaIcon from './icons/kafka.svg';
+import mongodbIcon from './icons/mongodb.svg';
+import mysqlIcon from './icons/mysql.svg';
+import postgresqlIcon from './icons/postgresql.svg';
+import redisIcon from './icons/redis.svg';
+import websocketIcon from './icons/websocket.svg';
+import javaIcon from '../../shared/AgentIcon/icons/java.svg';
+
+const defaultTypeIcons: { [key: string]: string } = {
+  cache: databaseIcon,
+  db: databaseIcon,
+  ext: globeIcon,
+  external: globeIcon,
+  messaging: documentsIcon,
+  resource: globeIcon,
+};
+
+const typeIcons: { [key: string]: { [key: string]: string } } = {
+  aws: {
+    servicename: awsIcon,
+  },
+  db: {
+    cassandra: cassandraIcon,
+    elasticsearch: elasticsearchIcon,
+    mongodb: mongodbIcon,
+    mysql: mysqlIcon,
+    postgresql: postgresqlIcon,
+    redis: redisIcon,
+  },
+  external: {
+    graphql: graphqlIcon,
+    grpc: grpcIcon,
+    websocket: websocketIcon,
+  },
+  messaging: {
+    jms: javaIcon,
+    kafka: kafkaIcon,
+  },
+  template: {
+    handlebars: handlebarsIcon,
+  },
+};
+
+export const defaultIcon = defaultIconImport;
+
+export function getSpanIcon(type?: string, subtype?: string) {
+  if (!type) {
+    return;
```

Comment:
We don't want to return a default icon in this case? 
Btw. I though `span.type` was always specified. Is that not the case?

Reply:
I'm aggregating on `span.destination.service.resource` only, so I can't guarantee that (due to needing to match with span events). But I will add a top_metrics aggregation on span.type and span.subtype to make sure `span.type` is always there as well.

Reply:
`span.type` is not available on span metrics, so I can't guarantee that it will actually be there (it should be in most cases, if we find a span event with the same `span.destination.service.resource`).